### ISSUE TITLE
`code-insiders --remote wsl+Ubuntu` results in weird state on Mac

### ISSUE THREAD

**Issue by JacksonKearl** [microsoft/vscode#91455]
Ref #91227

When I run that on Mac, it correctly gives an error that WSL extension is not installed and prompts me to install and reload. However after reloading I get an error that "WSL is only supported on Windows". I can navigate to the extension page for WSL and see that it's installed. Then, when trying to uninstall, I also get that error. I have to disconnect from the remote and then uninstall for it to go away.

**Comment by JacksonKearl** [comment]
It looks like the error on uninstall is actually not related to it not being available for mac, no extensions can be uninstalled when in that state. I do think disallowing `--remote wsl.*` on mac would make sense.

**Comment by aeschli** [comment]
> hen trying to uninstall, I also get that error

I think that's a bug we should look at. @sandy081 

Other than that I think it's not a common scenario that a Mac user would receive a WSL URI and  try to open it or call `code --remote wsl+Ubuntu`. We just do that do that for testing here.

**Comment by sandy081** [comment]
Yes, extensions actions do not work when a window is disconnected. Moving this remote repo.
